WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Me.swift

```diff
@@ -26,14 +27,10 @@ extension BlogDetailsViewController {
 
     @objc private func updateGravatarImage(_ notification: Foundation.Notification) {
         guard let userInfo = notification.userInfo,
-            let email = userInfo["email"] as? String,
-            let image = userInfo["image"] as? UIImage,
-            let url = AvatarURL.url(for: email),
-            let gravatarIcon = image.gravatarIcon(size: Metrics.iconSize) else {
-                return
+              let image = userInfo["image"] as? UIImage,
+              let gravatarIcon = image.gravatarIcon(size: Metrics.iconSize) else {
+            return
         }
-
-        ImageCache.shared.setImage(image, forKey: url.absoluteString)
```

Comment:
I'd keep the code that updates the cache. Uploading doesn't update the client side cache automatically. And the server side doesn't immediately start serving the new avatar, it can take like a couple of minutes. Until then, the client can update its own cache so it doesn't try to bring in the outdated avatar from server. 

Just make sure the caching key is exactly the same with the url that's used for fetching it for this UI component.

Reply:
I'm not using the memory cache in this scenario, but I probably should and there are [other parts](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/Modules/Sources/WordPressUI/Extensions/UIImageView%2BGravatar.swift) of the app that do.

I would probably avoid using app's shared memory cache (`ImageCache.shared`) for avatar caching as keeps images in cache indefinitely until the app is terminated.

What would you recommend, @pinarol?

Reply:
You should update `ImageCache.shared` whenever a new avatar is uploaded. Because it is the `ImageCache.shared` that the Gravatar SDK uses for avatar fetching operations (just search for "ImageCache.shared as? GravatarImageCaching"). If you don't update the cache here, the old avatar might remain in these UI elements, because when they try to re-fetch the image they'll bring the older one. I didn't test it but I expect this causes such a bug in the app. 

Whether using a memory cache or a different kind of cache is another topic we can discuss but here I am just saying you might be causing a bug. 🤔

Reply:
Looks like this code will have to be updated now because [WordpressUICacheAdapter](https://github.com/wordpress-mobile/WordPress-iOS/blob/80439eddcabf98301d18cabd5bfb446255867a11/WordPress/Classes/Utility/Media/MemoryCache.swift#L140) uses its`setEntry` and `getEntry` methods and is _not_ using `setImage`.

I pushed a new [commit](https://github.com/wordpress-mobile/WordPress-iOS/pull/23534/commits/f61782319cdb095e0cbfe797387403a913327993) adding new `GravatarImageService` to remove the code duplication, but I couldn't quite implement what I needed using the `Gravatar.ImageCaching` protocol: the images are currently blinking during the reload, and I think there are similar reports about some other parts of the app.

https://github.com/user-attachments/assets/e7aa0a02-2863-40e6-b97e-1b522f402f53

@pinarol is there a way `Gravatar` could provide a way to access images from its memory cache synchronously so that the app could its views without blinking?

I'm also a bit unsure about the intention behind the `CacheEntry` design. From the consumer perspective, I need to access memory cached images synchronously including during the reload. It does temporary set the entries to `inProgress` during the reload, doesn't it?